Xml model and validation for bpmn conditional events

berkaycanbc opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

Same PR as https://github.com/camunda/camunda/pull/41320. See the reason that I need to create a new PR here https://github.com/camunda/camunda/pull/41320#issuecomment-3575237466.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #40939

berkaycanbc committed:
feat: add XML model and validation for BPMN conditional events

berkaycanbc committed:
test: add validation and tests for conditional events in BPMN

berkaycanbc committed:
feat: add condition expression to revapi and make minor adjustment

berkaycanbc committed:
feat: add design time validation for condition

berkaycanbc committed:
feat: check duplicate start events with the same condition

lzgabel commented on zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ProcessConditionalStartEventValidator.java:
```diff
@@ -0,0 +1,100 @@
+    // event subprocess start events
+    final Collection<StartEvent> eventSubprocessStartEvents =
+        process.getChildElementsByType(SubProcess.class).stream()
+            .map(sub -> sub.getChildElementsByType(StartEvent.class))
```
:wrench: @berkaycanbc I think we should only need event sub-processes, please exclude embedded sub-processes.

```
process.getChildElementsByType(SubProcess.class).stream()
            .filter(SubProcess::triggeredByEvent) // filter event sub-processes
            .map(sub -> sub.getChildElementsByType(StartEvent.class))
            .flatMap(Collection::stream)
            .toList();
```

berkaycanbc committed:
refactor: remove redundant filter

lzgabel reviewed (approved):
🚀 @berkaycanbc Thanks. Please check my comment before merging.

georgios-goulos reviewed (approved):
LGTM. Feel free to merge once all comments have been addressed

berkaycanbc committed:
feat: only validate event subprocess for duplication

And add a test to verify embedded subprocess cannot have a conditional
start event.
